motors/tacho_motor_class.c

```diff
@@ -721,16 +861,17 @@ static ssize_t speed_sp_store(struct device *dev, struct device_attribute *attr,
 			      const char *buf, size_t size)
 {
 	struct tacho_motor_device *tm = to_tacho_motor(dev);
-	int err, speed;
-
-	if (!(BIT(TM_SPEED_REGULATION_ON) & tm->ops->get_speed_regulations(tm->context)))
-		return -EOPNOTSUPP;
+	int err, speed, max_speed;
 
 	err = kstrtoint(buf, 10, &speed);
 	if (err < 0)
 		return err;
 
-	if (speed > 2000 || speed < -2000)
+	err = tm->ops->get_max_speed(tm->context, &max_speed);
+	if (err < 0)
+		return err;
+
+	if (abs(speed) > max_speed)
```

Comment:
Well, you get into an area where you need to return a reasonable error for when the value is too high - previously it was a fixed 2000.

I think we should set `max_speed` to a higher value instead of allowing `speed_sp` to be a higher value. Or maybe we can allow `speed_sp` to be 120% of `max_speed`